### PR TITLE
better UEFI firmware handling: check existence, Fedora support

### PR DESCRIPTION
**Pull request by AdamWill**
There's a couple of problems with how UEFI firmware is handled
at present. If you pass the firmware location as UEFI_BIOS, the
check that the file actually exists gets skipped, because of
how things are ordered. Also, it will try and find a UEFI
firmware for you on x86_64 if BIOS and UEFI_BIOS aren't set,
but it only knows one possible location, the one in the package
for openSUSE. The packages from https://kraxel.org, commonly
used on Fedora, place a UEFI firmware somewhere else; this adds
support for that location (and makes it easy to support other
known locations as we find them). Finally, I made it possible
to pass an absolute path rather than one relative to
/usr/share/qemu, if you like. This also puts almost all the
handling in one place, leaving only the actual addition of the
qemu param to happen later (after the @params array is defined).

### CODE REVIEW

File: backend/qemu.pm [hunk 1]
```diff
@@ -216,10 +216,36 @@ sub start_qemu {
...
-    if ($vars->{BIOS} && !-e '/usr/share/qemu/' . $vars->{BIOS}) {
+    if ($vars->{UEFI}) {
+        # XXX: compat with old deployment
+        $vars->{BIOS} //= $vars->{UEFI_BIOS};
+    }
+
+    if ($vars->{BIOS} && $vars->{BIOS} !~ /^\//) {
+        # Non-absolute paths are assumed relative to /usr/share/qemu
+        $vars->{BIOS} == '/usr/share/qemu/' . $vars->{BIOS};
+    }
```
Comment (coolo) [inline backend/qemu.pm]: == ? unless this is yet another perl trick this must be =

**Comment by coolo** [comment]
otherwise fine with me, but I'm far from an UEFI expert

**Comment by nilxam** [comment]
Looks fine to me. It's default OVMF firmware(without specific key, if pure_uefi does this meaning) generated from EDK2 source, we're renamed them in OVMF firmwares.

File: backend/qemu.pm (hunk 1 above)
Comment (AdamWill) [inline backend/qemu.pm]: yes, sorry, I had some kinda virus yesterday - did this in another place too, but caught that one as I hit it in testing...I didn't test this bit.

**Pull request merged**